RIOT-OS/RIOT
# Atmega8 initial support

### Contribution description

The €1 board!

- boards/atmega8: copied from atmega328p
- cpu/atmega8: copied from atmega328p
- cpu/atmega_common: encountered some surprises in register organization...
- cpu/avr8_common: no system clock prescaler, no avr4.ld, no jmp instruction


### Testing procedure

The big challenge of this port! With only 8 Kbyte of flash and 1 Kbyte of RAM, hardly any test fits without modification, except periph/timer.

Therefore, in a second commit, I provide the streamlined test battery that I used (not mergeable, I believe).

Surprising things can be found, such as THREAD_STACKSIZE_IDLE=64 and THREAD_STACKSIZE_MAIN=128 for the SPI!

For the I2C, it's a bit particular. It's a use case of a 4x20 LCD screen connected through an I/O port expander (the only I2C thing I had on hand). It uses a slightly hacked version of the hd44780 driver because xtimer didn't fit...

There's also the shell, reduced as much as possible, which I ended up keeping only for the gpio test.

Regarding the hardware, I used this kind of board:
https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Ftse4.mm.bing.net%2Fth%3Fid%3DOIP.CMKk4Bbtbj7ZMmNHdHx74wHaHa%26pid%3DApi&f=1&ipt=26e53f5395ddeb5f231a2f9a6cdd1de7dc97e3e4e41f1fe6ead63fe8b8b56bfd&ipo=images

a usbasp programmer, and an stlink as a serial adapter. I used an oscilloscope to visualize the state of the ports for the gpio test, the SPI output signals (MISO not tested), and the PWM. For the ADC, I simply connected each input to ground to see if the value dropped to 0.

### Issues/PRs references
Split into 5:
- #19751 
- #19752
- #19753
- #19754
- #19755
